src/Evolution/DiscontinuousGalerkin/InboxTags.hpp

```diff
@@ -0,0 +1,126 @@
+#include <tuple>
+#include <type_traits>
+#include <utility>
+
+#include "DataStructures/FixedHashMap.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Domain/Structure/MaxNumberOfNeighbors.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Parallel/InboxInserters.hpp"
+#include "Time/TimeStepId.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace evolution::dg::Tags {
+/*!
+ * \brief The inbox tag for boundary correction communication and DG-subcell
+ * ghost zone cells.
+ *
+ * The first element in the tuple is the mesh of the neighboring element's
+ * face (not the mortar mesh!), the second is the variables at the ghost zone
+ * cells for finite difference/volume reconstruction, and the third is the data
+ * on the mortar needed for computing the boundary corrections (e.g. fluxes,
+ * characteristic speeds, conserved variables). he fourth element of the tuple
+ * is the TimeStepId beyond which the boundary terms are no longer valid, when
+ * using local time stepping.
+ *
+ * The TimeStepId is the neighboring element's next time step. When using local
+ * time stepping, the neighbor's boundary data is valid up until this time,
+ * which may include multiple local time steps. By receiving and storing the
+ * neighbor time step, the local element knows whether or not it should remove
+ * boundary data and expect new data to be sent from the neighbor.
+ *
+ * The ghost cell data (second element of the tuple) will be valid whenever
+ * DG-subcell scheme is being used. Whenever a DG-subcell scheme is being used,
+ * elements using DG and not FD/FV always send both the ghost cells and boundary
+ * correction data together. Elements using FD/FV send the ghost cells first
+ * followed by the boundary correction data once the element has received all
+ * neighbor ghost cell data. Note that the second send/receive only modifies the
+ * flux and the TimeStepId used for the flux validity range.
+ *
+ * When only a DG scheme (not a DG-subcell scheme) is used the ghost cell data
+ * will never be valid.
+ *
+ * In the DG-subcell scheme this tag is used both for communicating the ghost
+ * cell data needed for the FD/FV reconstruction step and the data needed for
+ * the boundary corrections. An element using a DG scheme will send both the
+ * ghost cells and the boundary correction terms simultaneously, while an
+ * element using FD/FV will first send the ghost cell data. The element
+ * receiving the ghost cells will store the data in this inbox tag and be able
+ * to do reconstruction once all neighboring ghost cells are received. In the
+ * mean time, as boundary correction data is received from neighbors it is added
+ * to this inbox tag. The inbox tag is not erased until after the boundary
+ * correction computation is complete. The reason for this design choice is that
+ * it is mostly the number of communications that dominates cost, not the size
+ * of the communication.
```

Comment:
I'm still not 100% on board with this paragraph. It starts by describing how data is received, but then the jump to reducing the number of comms is not obvious. What exactly is the choice you've made to favor larger comms in order to reduce the number of (reply in Github for now)?

Reply:
We need to send both pieces of data regardless. If we do 1 comm we save on the overhead of initiating a comm. Also, experimentally reducing the number of comms, even if it increases the comm size, is more efficient

Reply:
Okay, I think what I found confusing is your phrase "this design choice" after discussing so many things in the paragraph... to me this made it hard to know which aspect "this" was talking about. What do you think of adding a parenthetical to the last sentence, perhaps like this:

The reason for this design choice (i.e., combining the sent data into a single combination when the element uses DG) is that it is mostly the number of communications that dominates cost, not the size of the communication.

?

Reply:
What if I replace "this" by the thing in the parenthetical?

Reply:
👍